crates/circuits/src/builder/constraint_system.rs

```diff
@@ -110,13 +110,26 @@ impl<'arena> ConstraintSystemBuilder<'arena> {
 		direction: FlushDirection,
 		channel_id: ChannelId,
 		count: usize,
-		oracle_ids: impl IntoIterator<Item = OracleId> + Clone,
+		oracle_ids: impl IntoIterator<Item = OracleOrConst<F>> + Clone,
 		multiplicity: u64,
 	) -> anyhow::Result<()>
 	where
 		U: PackScalar<BinaryField1b>,
 	{
-		let n_vars = self.log_rows(oracle_ids.clone())?;
+		//We assume there is atleast one non constant in the collection of oracle ids.
+		let non_const_oracles: Vec<usize> = oracle_ids
+			.clone()
+			.into_iter()
+			.filter_map(|id| match id {
+				OracleOrConst::Oracle(oracle_id) => Some(oracle_id),
+				OracleOrConst::Const {
```

Comment:
nit: `OracleOrConst::Const { .. }`

Reply:
If you change it to `OracleOrConst::Const { .. }`. cargo fmt leaves it alone. I just tried.